android/PlayKit/app/src/main/java/io/streamroot/dna/samples/playkit/DynamicBufferLoadControl.java

```diff
@@ -0,0 +1,75 @@
+package io.streamroot.dna.samples.playkit;
+
+import com.kaltura.android.exoplayer2.LoadControl;
+
+import java.lang.reflect.Field;
+import java.util.concurrent.TimeUnit;
+
+public class DynamicBufferLoadControl {
+    private LoadControl mLoadControl;
+
+    private long mMinBufferUs = -1;
+    private boolean mIsSameMinMax = false;
+    private boolean mIsAVTargetSplit = false;
+
+    DynamicBufferLoadControl(LoadControl loadControl) {
+        mLoadControl = loadControl;
+        reflectLoadControl();
+    }
+
+    private Long getLoadControlPropertyLong(LoadControl loadControl, String propertyName) {
+        try {
+            Field loadControlField = loadControl.getClass().getDeclaredField(propertyName);
+            loadControlField.setAccessible(true);
+            return loadControlField.getLong(loadControl);
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    private boolean setLoadControlPropertyLong(LoadControl loadControl, String propertyName, long value) {
+        try {
+            Field loadControlField = loadControl.getClass().getDeclaredField(propertyName);
+            loadControlField.setAccessible(true);
+            loadControlField.setLong(loadControl, value);
+            return true;
+        } catch (Exception e) {
+            return false;
+        }
+    }
+
+    private void reflectLoadControl() {
+        Long reflectedMinBuffer = getLoadControlPropertyLong(mLoadControl, "minBufferUs");
+        if (reflectedMinBuffer == null) {
+            mIsAVTargetSplit = true;
+            Long reflectedMaxBuffer = getLoadControlPropertyLong(mLoadControl, "maxBufferUs");
+            Long reflectedMinVideoBuffer = getLoadControlPropertyLong(mLoadControl, "minBufferVideoUs");
+            if (reflectedMaxBuffer != null && reflectedMinVideoBuffer != null) {
+                if (reflectedMinVideoBuffer.equals(reflectedMaxBuffer)) {

```

Comment:
can this be resolved to a oneliner ternary just for readability? I dont know what ios kotlin practices about that

Reply:
If this is Kotlin, this is based on Kaltura's Android basic player sample which is in Java 😢

Reply:
Oh my bad! 🤣 totally out of subject here